src/backend/downloadmanager/downloadqueue.ts

```diff
@@ -340,6 +361,28 @@ function processNotification(element: DMQueueElement, status: DMStatus) {
   }
 }
 
+async function updateQueueElementParam<T extends keyof DMQueueElement>(
+  element: DMQueueElement,
+  prop: T,
+  value: DMQueueElement[T]
+) {
+  const queue: DMQueueElement[] = downloadManager.get('queue', [])
+
+  const index = queue.findIndex(
+    (el) => el.params.appName === element.params.appName
+  )
+  if (index !== -1) {
+    queue[index][prop] = value
+    if (currentElement) {
+      currentElement[prop] = value
+    }
```

Comment:
This is a side effect not really tied to what this function is for. What do you think about adding a `updateActiveQueueElement` function?

Reply:
Done